### PR TITLE
Fix Error Handling / Add Validation

### PR DESCRIPTION
**Pull request by ceanela**
Adding appropriate error handling and shape for client consumption

- [x] Display errors from `/create-account` process
- [x] Display errors from `/transfer` process
- [x] Display errors from `/balance` fetching
- [x] Display errors from `/transactions` fetching
- [x] Display errors when EOSD is offline

screen shot 2017-10-03 at 14 06 00

### CODE REVIEW

File: src/containers/CreateAccount/reducer.js [hunk 1]
```diff
@@ -114,10 +114,10 @@ function succeedPostEmail({ email }) {
   };
 }
 
-function failNewAccount({ user }) {
```
Comment (ceanela) [inline src/containers/CreateAccount/reducer.js]: when copypasta goes wrong...

File: src/containers/CreateAccount/reducer.js [hunk 2]
```diff
@@ -226,14 +230,13 @@ export function tryCreatePhone({ phone }) {
 
     return fetch('http://localhost:4000/api/account/new/', request)
       .then(async (response) => {
-        const clone = response.clone();
```
Comment (ceanela) [inline src/containers/CreateAccount/reducer.js]: removing these horrible .clone() calls

File: src/containers/CreateAccount/reducer.js [hunk 3]
```diff
@@ -257,22 +260,16 @@ export function tryCreateAccount(values) {
...
           dispatch(succeedNewAccount(user));
-
-          // history.push('/permissions');
-          // window.location.reload();
-
           return response;
         }
 
-        return Promise.reject(clone.json());
+        const errors = await response.json();
```
Comment (ceanela) [inline src/containers/CreateAccount/reducer.js]: in the future, i'd like to see a better way to handle errors, however for this PR, this is just an MVP for error handling

File: src/containers/LoginContainer/index.js [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
-  errorMessage,
-});
 
 const mapDispatchToProps = (dispatch, { history }) => ({
-  async onSubmit(values) {
-    await tryLogin(values)(dispatch);
-    dispatch(getBalance(values.account_name));
-    history.push('/');
+  async callAPI(values) {
+    return tryLogin(values)(dispatch)
```
Comment (ceanela) [inline src/containers/LoginContainer/index.js]: many many different shapes of `try[action]` some are thunks, some take dispatch in the parameter, etc., this needs to be resolved.

File: src/containers/TransferContainer/reducer.js [hunk 5]
```diff
@@ -95,7 +95,6 @@ export function tryPostTransaction({ memo, ...transaction }) {
             id: data.transaction_id,
           };
 
-          console.log('Transaction response \n\n\n', data);
```
Comment (ceanela) [inline src/containers/TransferContainer/reducer.js]: stop `console.log`-ing calls, that are logged with redux-logger

File: src/middleware/transactions.js [hunk 6]
```diff
@@ -15,9 +18,37 @@ const balance = store => next => (action) => {
     case 'SUCCEED_LOGIN':
     case 'SUCCESS_POST_TRANSACTION':
       dispatch(tryGetTransactions({ account_name }));
```
Comment (ceanela) [inline src/middleware/transactions.js]: again `try[action]` call is accepting data here, we need to get in sync with how we're going to be making calls to the server. i recommend to move that logic out of the reducer (really reducer + actions + types file) to middleware, where it's more appropriate.

Comment (ceanela) [reply src/middleware/transactions.js]: also, we should try to keep any `try[action]` dispatching out of middleware, it is anti-pattern

File: src/containers/CreateAccount/reducer.js (hunk 3 above)
Comment (ceanela) [reply src/containers/CreateAccount/reducer.js]: This is also lots and lots of boilerplate to do the same thing after a request

File: .env [hunk 7]
```diff
@@ -1,2 +1,3 @@
 # Endpoint for API calls into EOS platform/network
-EOS_NODE_ENDPOINT=http://127.0.0.1:8888
\ No newline at end of file
+EOS_NODE_ENDPOINT=https://eosd.objectcomputing.com
+PROXY_ENDPOINT=http://localhost:4400/
```
Comment (datchley) [inline .env]: Realizing I added the `.env` file to the application, typically the `.env` file is ignored and not checked into github.  We should probably determine how we want to manage configuration, letting the user override that with their own endpoints/ports, etc.

**Comment by ceanela** [comment]
Sure thing!

**Pull request merged**